src/getters/crunchbase.py

```diff
@@ -0,0 +1,114 @@
+"""
+discovery_utils.getters.crunchbase.py
+
+Module for easy access to downloaded CB data on S3.
+
+"""
+import pandas as pd
+import boto3
+import logging
+import re
+
+
+# Define global variables
+BUCKET_NAME = "discovery-iss"
+S3_PATH = "data/crunchbase/"
+AWS_ACCESS_KEY =
+AWS_SECRET_KEY = 
+FILE_NAMES = [
+    "acquisitions.csv",
+    "category_groups.csv",
+    "checksum.csv",
+    "degrees.csv",
+    "event_appearances.csv",
+    "events.csv",
+    "funding_rounds.csv",
+    "funds.csv",
+    "investment_partners.csv",
+    "investments.csv",
+    "investors.csv",
+    "ipos.csv",
+    "jobs.csv",
+    "org_parents.csv",
+    "organization_descriptions.csv",
+    "organizations.csv",
+    "people_descriptions.csv",
+    "people.csv"
+]
+
+
+# Initialize S3 client
+S3 = boto3.client(
+    "s3", aws_access_key_id=AWS_ACCESS_KEY, aws_secret_access_key=AWS_SECRET_KEY
+    )
+
+
+# Building block functions - not to be used directly
+
+def list_s3_directories():
+    """List S3 directories under the specified prefix."""
+    # List all objects with the specified prefix
+    objects = S3.list_objects_v2(Bucket=BUCKET_NAME, Prefix=S3_PATH, Delimiter='/')
+
+    # Extract the directory names from the CommonPrefixes list
+    directories = [prefix['Prefix'] for prefix in objects.get('CommonPrefixes', [])]
+
+    return directories
+
+def extract_timestamp(directory):
+    """ Use regular expression to extract the timestamp from a directory"""
+    match = re.search(r'\d{8}_\d{6}', directory)
+    if match:
+        timestamp_str = match.group()
+        return timestamp_str
+    else:
+        return None
+    
+def directory(position) -> str:
+    """Directory name for a given snapshot of CB data."""
+    
+    # Get the list of directories
+    directories = list_s3_directories()
+    
+    # Get a list of timestamps
+    timestamps = []
+    for directory in directories:
+        timestamps.append(extract_timestamp(directory))
+    
+    # Sort the timestamps in descending order
+    sorted_timestamps = sorted(timestamps, reverse=True)
+    
+    # Return the most recent directory
+    for directory in directories:
+        if sorted_timestamps[position] in directory:
+            return directory
+
+def get_crunchbase_table(file_name, version) -> pd.DataFrame:
+    """Table with named CB data"""
+    s3_key = S3_PATH + version + file_name
+    logging.info(f"Fetching data from S3 key: {s3_key}")
+
+    # Read the file contents from S3
+    response = S3.get_object(Bucket=BUCKET_NAME, Key=s3_key)
+    data = response["Body"].read().decode("utf-8")
+    
+    return pd.read_csv(data)
```

Comment:
Just a note that almost all functions will be useful also for other getters in the future. OK to keep it like this for now, but we'll probably want to eventually factor them out.

Also, in this case, I think we could use the [single underscore notation](https://realpython.com/python-double-underscore/) for the functions that get directory list from s3, extracts timestamps etc as they are not supposed to be called by the user, but only used "internally" by other functions

Solomon @sqr00t, what do you think about these points? You'll definitely have some insights on how to best organise the utils.

Reply:
Yep! Single underscores are the way for semi-private functions that are not intended for top-level usage. Also, I would move some of these functions to `src/utils/`, especially the ones that are not getters (i.e. prefixed with get_)

Reply:
Thanks for the suggestions @beingkk and @sqr00t. I've read up on the notation conventions and added single underscores to the non-public functions.

@sqr00t - I'd like to confirm with you which ones should be moved, and would be great to understand your logic as to why.

Reply:
I'd say everything that does not eagerly "get" the data should live in utils.

Examples include functions evaluating a complete s3 path, or getting directories, since these are agnostic and generalisable across pipelines.